modules/distribution/src/bin/openssl-tls.sh

```diff
@@ -0,0 +1,320 @@
+        # Check if libssl-dev is installed
+        if [ -f "/etc/debian_version" ]; then
+            if ! dpkg -s libssl-dev > /dev/null 2>&1; then
+                echo "libssl-dev is not installed. Please install it before running this script."
+                exit 1
+            fi
+        else
+            if ! rpm -q openssl-devel > /dev/null 2>&1; then
+                echo "openssl-devel is not installed. Please install it before running this script."
+                exit 1
+            fi
+        fi
+        # Check if libapr1-dev is installed
+        if [ -f "/etc/debian_version" ]; then
+            if ! dpkg -s libapr1-dev > /dev/null 2>&1; then
+                echo "libapr1-dev is not installed. Please install it before running this script."
+                exit 1
+            fi
+        else
+            if ! rpm -q apr-devel > /dev/null 2>&1; then
+                echo "apr-devel is not installed. Please install it before running this script."
+                exit 1
+            fi
+        fi
+    fi
+    check_dependencies apr-1-config
+fi
+
+TMP_DIR="$CARBON_HOME"/openssl-tmp
+if [ -d "$TMP_DIR" ]; then
+    rm -rf "$TMP_DIR"
+fi
+mkdir -p "$TMP_DIR"
+
+# =============================== Install OpenSSL ================================================
+
+if [ $BUILD_OPENSSL = true ]; then
+    OPENSSL_INSTALL_DIR="$CARBON_HOME/native/openssl"
+elif [ "$(uname)" = "Darwin" ]; then
+    if [ "$(uname -m)" = "arm64" ]; then
+        # For M1 (ARM) architecture
+        OPENSSL_INSTALL_DIR=$(ls -d '/opt/homebrew/Cellar/openssl@3/'*)
+    else
+        # For Intel (x86_64) architecture
+        OPENSSL_INSTALL_DIR=$(ls -d '/usr/local/Cellar/openssl@3/'*)
+    fi
+fi
+
+if [ $BUILD_OPENSSL = true ] && [ ! -d "$OPENSSL_INSTALL_DIR" ]; then
+    install_openssl "$TMP_DIR" "$OPENSSL_INSTALL_DIR"
+else
+    echo "OpenSSL is already installed"
+fi
+
+# =============================== Install OQS Provider ============================================
+
+if [ $BUILD_PQCLIB = true ]; then
+    if [ -d "$OPENSSL_INSTALL_DIR/lib64" ]; then
+        OPENSSL_LIB_DIR="$OPENSSL_INSTALL_DIR/lib64"
+    else
+        OPENSSL_LIB_DIR="$OPENSSL_INSTALL_DIR/lib"
+    fi
+
+    # Check if OQS Provider is already installed
+    if [ ! -f "$CARBON_HOME/lib/oqsprovider.$LIBEXT" ]; then
+        if [ $BUILD_OPENSSL = true ] || [ "$(uname)" = "Darwin" ]; then
+            install_oqs_provider "$TMP_DIR" "$OPENSSL_INSTALL_DIR" "$OPENSSL_LIB_DIR/libcrypto.$LIBEXT"
+        else
+            install_oqs_provider "$TMP_DIR"
+        fi
+        mv "$TMP_DIR/oqs-provider-$OQS_PROVIDER_VERSION/_build/lib/oqsprovider.$LIBEXT" "$CARBON_HOME/lib"
+    else
+        echo "OQS Provider is already installed"
+    fi
+fi
+
+# =============================== Install APR ====================================================
+
+if [ $BUILD_OPENSSL = true ]; then
+    rm -rf "$CARBON_HOME/native/apr"
+    install_apr "$TMP_DIR" "$CARBON_HOME/native/apr"
```

Comment:
Why is the APR installation forced here? Openssl and oqs installations above don't follow this approach.

Reply:
In APR, the absolute path is stored during installation. Therefore need to force installation just like Tomcat native